src/clients/transports/fallback.ts

```diff
@@ -60,4 +100,104 @@ export function fallback(
         ),
       },
     )
+
+    if (rank) {
+      const rankOptions = typeof rank === 'object' ? rank : {}
+      rankTransports({
+        chain,
+        interval: rankOptions.interval ?? pollingInterval,
+        latencyWeight: rankOptions.latencyWeight,
+        onTransports: (transports_) => (transports = transports_),
+        sampleCount: rankOptions.sampleCount,
+        stabilityWeight: rankOptions.stabilityWeight,
+        timeout: rankOptions.timeout,
+        transports,
+      })
+    }
+
+    return transport
+  }
+}
+
+type SampleData = [latency: number, success: number]
+type Sample = SampleData[]
+
+export function rankTransports({
+  chain,
+  interval = 4_000,
+  latencyWeight = 0.3,
+  onTransports,
+  sampleCount = 10,
+  stabilityWeight = 0.7,
+  timeout = 1_000,
+  transports,
+}: {
+  chain?: Chain
+  interval: number
+  latencyWeight?: number
+  onTransports: (transports: Transport[]) => void
+  sampleCount?: number
+  stabilityWeight?: number
+  timeout?: number
+  transports: Transport[]
+}) {
+  let samples: Sample[] = []
```

Comment:
If `SampleData` and `Sample` are only used internally, could move them inside the function so we don't accidentally use them anywhere else.

```suggestion
export function rankTransports({
  chain,
  interval = 4_000,
  latencyWeight = 0.3,
  onTransports,
  sampleCount = 10,
  stabilityWeight = 0.7,
  timeout = 1_000,
  transports,
}: {
  chain?: Chain
  interval: number
  latencyWeight?: number
  onTransports: (transports: Transport[]) => void
  sampleCount?: number
  stabilityWeight?: number
  timeout?: number
  transports: Transport[]
}) {
  type SampleData = [latency: number, success: number]
  type Sample = SampleData[]
  let samples: Sample[] = []
```